app/src/main/java/com/example/androidDeviceDetails/viewModel/location/LocationViewModel.kt

```diff
@@ -0,0 +1,128 @@
+package com.example.androidDeviceDetails.viewModel.location
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.view.View.GONE
+import android.widget.Toast
+import androidx.core.content.ContextCompat.getDrawable
+import com.example.androidDeviceDetails.R
+import com.example.androidDeviceDetails.adapters.LocationAdapter
+import com.example.androidDeviceDetails.base.BaseViewModel
+import com.example.androidDeviceDetails.databinding.ActivityLocationBinding
+import com.example.androidDeviceDetails.models.location.LocationItemViewHolder
+import com.github.davidmoten.geo.GeoHash.decodeHash
+import com.google.maps.android.ui.IconGenerator
+import org.osmdroid.util.GeoPoint
+import org.osmdroid.views.overlay.Marker
+import org.osmdroid.views.overlay.Overlay
+
+
+class LocationViewModel(private val binding: ActivityLocationBinding, val context: Context) :
+    BaseViewModel() {
+
+    private lateinit var cookedDataList: ArrayList<LocationItemViewHolder>
+
+    private fun toggleSortButton() {
+        if (binding.bottomLocation.sortButton.tag == "down") {
+            binding.bottomLocation.sortButton.tag = "up"
+            binding.bottomLocation.sortButton.setCompoundDrawables(
+                null, null, getDrawable(context, R.drawable.ic_arrow_upward), null
+            )
+
+        } else {
+            binding.bottomLocation.sortButton.tag = "down"
+            binding.bottomLocation.sortButton.setCompoundDrawables(
+                null, null, getDrawable(context, R.drawable.ic_arrow_downward), null
+            )
+        }
+    }
+
+    private fun onNoData() =
+        binding.root.post {
+            Toast.makeText(context, "No data on selected date", Toast.LENGTH_SHORT).show()
+        }
+
+
+    override fun sort(type: Int) {
+        binding.root.post {
+            (binding.bottomLocation.locationListView.adapter as LocationAdapter).sortView(type)
+        }
+        toggleSortButton()
+    }
+
+    private fun buildAdapterView() {
+        binding.root.post {
+            (binding.bottomLocation.locationListView.adapter as LocationAdapter).refreshList(
+                cookedDataList
+            )
+        }
+    }
+
+    private fun drawableToBitmap(drawable: Drawable): Bitmap? {
+        if (drawable is BitmapDrawable)
+            return drawable.bitmap
+        val bitmap = Bitmap.createBitmap(
+            drawable.intrinsicWidth, drawable.intrinsicHeight, Bitmap.Config.ARGB_8888
+        )
+        val canvas = Canvas(bitmap)
+        drawable.setBounds(0, 0, canvas.width, canvas.height)
+        drawable.draw(canvas)
+        return bitmap
+    }
+
+    private fun addPointOnMap() {
+        val overlays: MutableList<Overlay> = binding.mapView.overlays
```

Comment:
-  : MutableList<Overlay> is not required you can directly assign

Reply:
done